packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -30,19 +31,32 @@ function BlockListAppender( {
 	rootClientId,
 	canInsertDefaultBlock,
 	isLocked,
-	renderAppender: CustomAppender,
+	renderAppender: CustomAppender, // Deprecated.
+	appender,
 	className,
 	selectedBlockClientId,
 	tagName: TagName = 'div',
 } ) {
-	if ( isLocked || CustomAppender === false ) {
+	if ( undefined !== CustomAppender ) {
+		deprecated( 'renderAppender prop', {
+			since: '10.9',
+			alternative: 'appender prop',
+		} );
+	}
+
+	if (
+		isLocked ||
+		( undefined !== appender && ! appender ) ||
+		false === CustomAppender
+	) {
 		return null;
 	}
 
-	let appender;
-	if ( CustomAppender ) {
-		// Prefer custom render prop if provided.
-		appender = <CustomAppender />;
+	let blockAppender;
+	if ( undefined !== appender ) {
```

Comment:
As other falsey values are avoided by the earlier `if` statement, this could probably just be `if ( appender ) {`.

What do you think?

Reply:
Ah, yeah! Makes sense too! https://github.com/WordPress/gutenberg/pull/29911/commits/25317c9352acca0c96e7ef004ef721616a5f8bda